packages/ember-extension-support/tests/container_debug_adapter_test.js

```diff
@@ -1,52 +1,47 @@
 
-function boot() {
-  run(App, 'advanceReadiness');
-}
+moduleFor('Container Debug Adapter', class extends ApplicationTestCase {
+  constructor() {
+    super();
+    adapter = this.application.__deprecatedInstance__.lookup('container-debug-adapter:main');
+  }
 
-QUnit.module('Container Debug Adapter', {
-  setup() {
-    run(() => {
-      App = Application.create();  // ES6TODO: this comes from the ember-application package NOT ember-runtime.
-      App.toString = function() { return 'App'; };
-      App.deferReadiness();
-    });
-    boot();
-    run(() => {
-      appInstance = App.__deprecatedInstance__;
-      adapter = appInstance.lookup('container-debug-adapter:main');
+  get applicationOptions() {
+    return assign(super.applicationOptions, {
+      autoboot: true
     });
-  },
+  }
+
   teardown() {
     run(() => {
       adapter.destroy();
-      appInstance.destroy();
-      App.destroy();
-      App = appInstance = adapter = null;
     });
+
+    super.teardown();
   }
-});
 
-QUnit.test('the default ContainerDebugAdapter cannot catalog certain entries by type', function() {
-  equal(adapter.canCatalogEntriesByType('model'), false, 'canCatalogEntriesByType should return false for model');
-  equal(adapter.canCatalogEntriesByType('template'), false, 'canCatalogEntriesByType should return false for template');
-});
+  ['@test default ContainerDebugAdapter cannot catalog certain entries by type'](assert) {
+    assert.equal(adapter.canCatalogEntriesByType('model'), false, 'canCatalogEntriesByType should return false for model');
+    assert.equal(adapter.canCatalogEntriesByType('template'), false, 'canCatalogEntriesByType should return false for template');
+  }
 
-QUnit.test('the default ContainerDebugAdapter can catalog typical entries by type', function() {
-  equal(adapter.canCatalogEntriesByType('controller'), true, 'canCatalogEntriesByType should return true for controller');
-  equal(adapter.canCatalogEntriesByType('route'), true, 'canCatalogEntriesByType should return true for route');
-  equal(adapter.canCatalogEntriesByType('view'), true, 'canCatalogEntriesByType should return true for view');
-});
+  ['@test default ContainerDebugAdapter can catalog typical entries by type'](assert) {
+    assert.equal(adapter.canCatalogEntriesByType('controller'), true, 'canCatalogEntriesByType should return true for controller');
+    assert.equal(adapter.canCatalogEntriesByType('route'), true, 'canCatalogEntriesByType should return true for route');
+    assert.equal(adapter.canCatalogEntriesByType('view'), true, 'canCatalogEntriesByType should return true for view');
+  }
 
-QUnit.test('the default ContainerDebugAdapter catalogs controller entries', function() {
-  App.PostController = EmberController.extend();
-  let controllerClasses = adapter.catalogEntriesByType('controller');
+  ['@test default ContainerDebugAdapter catalogs controller entries'](assert) {
+    this.application.PostController = EmberController.extend()
```

Comment:
It seems that is not possible to use application registry for the controller since the actually implementation of `catalogEntriesByType` looks for properties in the app.